@ABLITZER/visualizeLearningResponses.m

```diff
@@ -1,110 +1,110 @@
-   if p > 0.05 % n.s.
-       text(2,0.25,'n.s.','FontSize',14);
-   elseif p < 0.05 % "*"
-       text(2,0.25,'*','FontSize',20);
-   elseif p < 0.01 % "**"
-       text(2,0.25,'**','FontSize',20);
-   elseif p < 0.001 % "***"
-       text(2,0.25,'***','FontSize',20);
-   elseif p < 0.0001 % "****"
-       text(2,0.25,'****','FontSize',20);
-   end
-   
-   % experiment group: compare mean of PIs between Baseline and Test
-   [~,p] = ttest(PIs(:,4),PIs(:,6));
-   line([4,6],[0.3,0.3],'Color',[0,0,0]);
-   if p > 0.05 % n.s.
-       text(5,0.25,'n.s.','FontSize',14);
-   elseif p < 0.05 % "*"
-       text(5,0.25,'*','FontSize',20);
-   elseif p < 0.01 % "**"
-       text(5,0.25,'**','FontSize',20);
-   elseif p < 0.001 % "***"
-       text(5,0.25,'***','FontSize',20);
-   elseif p < 0.0001 % "****"
-       text(5,0.25,'****','FontSize',20);
-   end
-   
-   
-
-end
\ No newline at end of file
+% plot PIs of an entire group to see whether there's
+% any statistical significance.
+%
+% INPUT:
+%   idxExpGroup: the index of experiment group data in FishGroup struct
+%   idxCtrlGroup: the index of control group data in FishGroup struct
+%   metricType: which metric the user wants to plot. (PItime, PIturn or
+%   PIshock?)
+%
+%   TODO: put contrain on pairing experiment data with control data
+%       currently, they are automatically in the same order due to
+%       alphabetic reason.
+function visualizeLearningResponses(obj,idxExpGroup,idxCtrlGroup,metricType)
+    if (nargin == 3) % default plot type is based on type
+        metricType = "time";
+    end
+    fishStack = obj.FishStack;
+    idxExpData = obj.FishGroups(idxExpGroup).Data;
+    idxCtrlData = obj.FishGroups(idxCtrlGroup).Data;
+
+    expData = fishStack(idxExpData);
+    ctrlData = fishStack(idxCtrlData);
+
+    % Pairing check (temporal)
+    if length(expData) ~= length(ctrlData)
+        error('Data from experiment group is not paired with the control group.');
+    end
+
+    qualThre = 0.95;
+    PIs = []; % Performance Index from control and experiment group
+    for i=1:length(expData)
+        % Eliminate fish whose dataQuality is lower than qualThre (0.95)
+        % once expData/ctrlData eliminated, both data eliminated
+        if (expData(i).Res.DataQuality < qualThre) || (ctrlData(i).Res.DataQuality < qualThre)
+            fprintf('Invalid Data:\n DataQuality is lower than 0.95.\n'); continue;
+        end
+        if contains(metricType,"time",'IgnoreCase',true)
+            PIdata = cat(2,ctrlData(i).Res.PItime.PIfish,expData(i).Res.PItime.PIfish);
+        elseif contains(metricType,"turn",'IgnoreCase',true)
+            PIdata = cat(2,ctrlData(i).Res.PIturn.PIfish,expData(i).Res.PIturn.PIfish);
+        end
+
+        PIs = cat(1,PIs, PIdata);
+    end
+
+    if isempty(PIs)
+        fprintf('No valid data to plot\n');
+        return
+    end
+
+    figure;
+
+    colors = 0.8*ones(6,3);%[0.5,0,0;0,0.5,0;0,0,0.5;0.5,0.5,0;0.5,0,0.5;0,0.5,0.5];
```

Comment:
It might be better to replace this part of plotting code with a sub-function since it's more modular.